src/main/java/com/microfocus/octane/plugins/rest/query/OctaneQueryBuilder.java

```diff
@@ -65,12 +65,16 @@ public OctaneQueryBuilder addQueryConditions(Collection<QueryPhrase> conditions)
     }
 
     public OctaneQueryBuilder addStartIndex(Integer startIndex) {
-        this.startIndex = startIndex;
+        if (startIndex != null) {
+            this.startIndex = startIndex;
+        }
         return this;
     }
 
-    public OctaneQueryBuilder addPageSize(int pageSize) {
-        this.pageSize = pageSize;
+    public OctaneQueryBuilder addPageSize(Integer pageSize) {
```

Comment:
Can't you handle this in a more elegant way? e.g. method overloading? I really don't like this null checks and mix of param types (primitive & object types, calling methods with null values for params)